backend/core/chunking/chunker.py

```
@@ -0,0 +1,1582 @@
+            pdf_reader = PyPDF2.PdfReader(file)
+            for page in pdf_reader.pages:
+                text += page.extract_text() or ""
+        return text
+    
+    def _extract_docx_text(self, file_path: Path) -> str:
+        doc = Document(file_path)
+        return "\n".join([paragraph.text for paragraph in doc.paragraphs])
+    
+    def _extract_markdown_text(self, file_path: Path) -> str:
+        with open(file_path, 'r', encoding='utf-8') as file:
+            html = markdown.markdown(file.read())
```

Ruff (0.15.9)

[error] Undefined name `markdown`

(F821)